// @discordjs/builders/dist/messages/formatters.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TimestampStyles = exports.time = exports.hyperlink = exports.hideLinkEmbed = exports.blockQuote = exports.quote = exports.strikethrough = exports.underscore = exports.bold = exports.italic = exports.inlineCode = exports.codeBlock = void 0;
function codeBlock(language, content) {
    return typeof content === 'undefined' ? `\`\`\`\n${language}\`\`\`` : `\`\`\`${language}\n${content}\`\`\``;
}
exports.codeBlock = codeBlock;
/**
 * Wraps the content inside an inline code.
 * @param content The content to wrap.
 * @returns The formatted content.
 */
function inlineCode(content) {
    return `\`${content}\``;
}
exports.inlineCode = inlineCode;
/**
 * Formats the content into italic text.
 * @param content The content to wrap.
 * @returns The formatted content.
 */
function italic(content) {
    return `_${content}_`;
}
exports.italic = italic;
/**
 * Formats the content into bold text.
 * @param content The content to wrap.
 * @returns The formatted content.
 */
function bold(content) {
    return `**${content}**`;
}
exports.bold = bold;
/**
 * Formats the content into underscored text.
 * @param content The content to wrap.
 * @returns The formatted content.
 */
function underscore(content) {
    return `__${content}__`;
}
exports.underscore = underscore;
/**
 * Formats the content into strikethrough text.
 * @param content The content to wrap.
 * @returns The formatted content.
 */
function strikethrough(content) {
    return `~~${content}~~`;
}
exports.strikethrough = strikethrough;
/**
 * Formats the content into a quote. This needs to be at the start of the line for Discord to format it.
 * @param content The content to wrap.
 * @returns The formatted content.
 */
function quote(content) {
    return `> ${content}`;
}
exports.quote = quote;
/**
 * Formats the content into a block quote. This needs to be at the start of the line for Discord to format it.
 * @param content The content to wrap.
 * @returns The formatted content.
 */
function blockQuote(content) {
    return `>>> ${content}`;
}
exports.blockQuote = blockQuote;
function hideLinkEmbed(url) {
    // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
    return `<${url}>`;
}
exports.hideLinkEmbed = hideLinkEmbed;
function hyperlink(content, url, title) {
    // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
    return title ? `[${content}](${url} "${title}")` : `[${content}](${url})`;
}
exports.hyperlink = hyperlink;
function time(timeOrSeconds, style) {
    if (typeof timeOrSeconds !== 'number') {
        timeOrSeconds = Math.floor((timeOrSeconds?.getTime() ?? Date.now()) / 1000);
    }
    return typeof style === 'string' ? `<t:${timeOrSeconds}:${style}>` : `<t:${timeOrSeconds}>`;
}
exports.time = time;
/**
 * The [message formatting timestamp styles](https://discord.com/developers/docs/reference#message-formatting-timestamp-styles) supported by Discord.
 */
exports.TimestampStyles = {
    /**
     * Short time format, consisting of hours and minutes, e.g. 16:20.
     */
    ShortTime: 't',
    /**
     * Long time format, consisting of hours, minutes, and seconds, e.g. 16:20:30.
     */
    LongTime: 'T',
    /**
     * Short date format, consisting of day, month, and year, e.g. 20/04/2021.
     */
    ShortDate: 'd',
    /**
     * Long date format, consisting of day, month, and year, e.g. 20 April 2021.
     */
    LongDate: 'D',
    /**
     * Short date-time format, consisting of short date and short time formats, e.g. 20 April 2021 16:20.
     */
    ShortDateTime: 'f',
    /**
     * Long date-time format, consisting of long date and short time formats, e.g. Tuesday, 20 April 2021 16:20.
     */
    LongDateTime: 'F',
    /**
     * Relative time format, consisting of a relative duration format, e.g. 2 months ago.
     */
    RelativeTime: 'R',
};
